integrations/tensorflow/e2e/tensorlist_test.py

```diff
@@ -66,50 +68,59 @@ def concat_with_tensorlist_stack(self, a, b):
     return ta.stack()
 
 
-@tf_test_utils.compile_module(TensorListModule)
 class TensorListTest(tf_test_utils.TracedModuleTestCase):
 
+  def __init__(self, methodName="runTest"):
+    super(TensorListTest, self).__init__(methodName)
+    self._modules = tf_test_utils.compile_tf_module(TensorListModule)
+
   def test_identity_through_tensorlist(self):
 
     def identity_through_tensorlist(module):
       module.identity_through_tensorlist(np.array(42., dtype=np.float32))
 
-    self.compare_backends(identity_through_tensorlist)
+    self.compare_backends(identity_through_tensorlist, self._modules)
 
   def test_add_through_tensorlist(self):
 
     def add_through_tensorlist(module):
-      module.add_through_tensorlist(
-          np.array(42., dtype=np.float32), np.array(43., dtype=np.float32))
+      module.add_through_tensorlist(np.array(42., dtype=np.float32),
+                                    np.array(43., dtype=np.float32))
 
-    self.compare_backends(add_through_tensorlist)
+    self.compare_backends(add_through_tensorlist, self._modules)
 
   def test_slice_first_element_with_from_tensor(self):
 
     def slice_first_element_with_from_tensor(module):
       module.slice_first_element_with_from_tensor(
           np.arange(STATIC_SIZE, dtype=np.float32))
 
-    self.compare_backends(slice_first_element_with_from_tensor)
+    self.compare_backends(slice_first_element_with_from_tensor, self._modules)
 
   def test_slice_first_element_with_from_tensor_high_rank(self):
 
     def slice_first_element_with_from_tensor_high_rank(module):
       module.slice_first_element_with_from_tensor_high_rank(
           tf_utils.ndarange([STATIC_SIZE, STATIC_SIZE]))
 
-    self.compare_backends(slice_first_element_with_from_tensor_high_rank)
+    self.compare_backends(slice_first_element_with_from_tensor_high_rank,
+                          self._modules)
 
   def test_concat_with_tensorlist_stack(self):
 
     def concat_with_tensorlist_stack(module):
-      module.concat_with_tensorlist_stack(
-          np.array(42., dtype=np.float32), np.array(43., dtype=np.float32))
+      module.concat_with_tensorlist_stack(np.array(42., dtype=np.float32),
+                                          np.array(43., dtype=np.float32))
 
-    self.compare_backends(concat_with_tensorlist_stack)
+    self.compare_backends(concat_with_tensorlist_stack, self._modules)
 
 
-if __name__ == "__main__":
-  if hasattr(tf, "enable_v2_behavior"):
+def main(argv):
+  del argv  # Unused
+  if hasattr(tf, 'enable_v2_behavior'):
     tf.enable_v2_behavior()
   tf.test.main()
+
+
+if __name__ == '__main__':
+  app.run(main)
```

Comment:
nit: is the app.run still needed in the new design?

Reply:
It's not necessary for most of the tests, but it's useful to have if in case you need to condition on flags (e.g. in `vision_model_test.py`). We also know that everyone running these tests already has `absl` from `tensorflow`. 

I'd vote to keep it to keep things uniform between the tests, but I'm fine either way!

Reply:
Keeping it consistent SGTM. Last thing we want is for copypasted code to not work due to some obscure flag parsing issue.